totoify_grafeas/totoifylib.py

```diff
@@ -0,0 +1,266 @@
+    """Returns the JSON string representation."""
+    if file_path:
+      with open(file_path, "w") as fp:
+        json.dump({
+            "resource": self.resource,
+            "noteName": self.note_name,
+            "kind": self.kind,
+            "intoto": {
+              "signatures": self.intoto["signatures"],
+              "signed": self.intoto["signed"].to_dict()
+            }
+          }, fp, indent=4)
+    else:
+      return json.dumps({
+          "resource": self.resource,
+          "noteName": self.note_name,
+          "kind": self.kind,
+          "intoto": {
+            "signatures": self.intoto["signatures"],
+            "signed": self.intoto["signed"].to_dict()
+          }
+        }
+      )
+
+  @staticmethod
+  def load(path):
+    """Given a path, loads the occurrence.json into GrafeasInTotoOccurrence
+    class.
+    """
+    with open(path, "r") as occ_f:
+      occ_json = json.load(occ_f)
+
+    grafeas_occurrence = GrafeasInTotoOccurrence()
+    grafeas_occurrence.intoto['signed'] = \
+        GrafeasLink(occ_json['intoto']['signed']['materials'],
+                    occ_json['intoto']['signed']['products'],
+                    occ_json['intoto']['signed']['command'],
+                    occ_json['intoto']['signed']['byproducts'],
+                    occ_json['intoto']['signed']['environment'])
+    grafeas_occurrence.intoto['signatures'] = occ_json['intoto']['signatures']
+    grafeas_occurrence.resource['uri'] = occ_json['resource']['uri']
+    grafeas_occurrence.noteName = occ_json['noteName']
+    return grafeas_occurrence
+
+  @staticmethod
+  def from_link(in_toto_link, step_name, resource_uri):
+    """Returns a GrafeasInTotoOccurrence class from an In-toto Metablock,
+    stepname, and resource uri."""
+    # TODO: set step_name from link if None
+    return GrafeasInTotoOccurrence(in_toto_link, step_name, resource_uri)
+
+  def to_link(self, step_name):
+    """Returns an in-toto link Metablock class from a GrafeasInTotoOccurrence
+    class.
+    """
+    materials = {}
+    products = {}
+    command = []
+    byproducts = {}
+    environment = {}
+
+    for item in self.intoto["signed"].materials:
+      materials[item["resource_uri"]] = item["hashes"]
+
+    for item in self.intoto["signed"].products:
+      products[item["resource_uri"]] = item["hashes"]
+
+    command = self.intoto["signed"].command
+
+    for key, value in self.intoto["signed"].byproducts.items():
+      if key == "custom_values":
+        continue
+      byproducts[key] = value
+
+    if "custom_values" in self.intoto["signed"].byproducts:
+      for key, value in \
+          self.intoto["signed"].byproducts["custom_values"].items():
+        if key == "return-value":
+          byproducts[key] = int(value)  # cast return-value back to int
+        else:
+          byproducts[key] = value
+
+    for key, value in self.intoto["signed"].environment.items():
+      if key == "custom_values":
```

Comment:
Maybe add a comment that we will handle this case below (and also why we handle it below and not here). Also, do we expect any other values?